components/ruby/lib/chef-licensing/license_key_fetcher/license_file/v3.rb

```diff
@@ -0,0 +1,38 @@
+require_relative "base"
```

Comment:
I had a feeling that why we should create sub-classes, Is it just to identify the versions of the file and differentiate the structure?  Can we accomplish this in one class? Some methods are duplication in both the classes then is it really achieving inheritance here?

Reply:
The different versions can have different implementation of the primary structure or expected structure and the base class for those methods simply raises not implemented error if the inherited class doesn't define it. The design was inspired by factory method https://refactoring.guru/design-patterns/factory-method/ruby/example

However, I have some refactoring to do which should make some of the method definition defined in v3 and v4 class to be included as part of the base class